apps/src/code-studio/pd/application_dashboard/detail_view_contents.jsx

```diff
@@ -258,6 +261,20 @@ export class DetailViewContents extends React.Component {
     }
   }
 
+  componentDidUpdate(prevProps) {
+    if (
+      prevProps.applicationData.update_emails_sent_by_system !==
+      this.props.applicationData.update_emails_sent_by_system
+    ) {
+      this.setState({
+        statuses: getApplicationStatuses(
+          this.props.viewType,
+          this.props.applicationData.update_emails_sent_by_system
+        )
+      });
+    }
+  }
```

Comment:
Instead of storing the rendered status list in state, I might just recompute it when needed - it's a fairly small operation.  That would remove this derived state logic entirely.

Reply:
done!